src/components/generic/BooleanField.js

```diff
@@ -0,0 +1,63 @@
+import styled from "styled-components";
+
+
+const Container = styled.div.attrs(
+    props => ({
+        style: {
+            margin:         `${ props.fontSize * 0.5 }rem` || `0.35rem`,
+            padding:        `-${ props.fontSize * 0.5 }rem` || `0.25rem`,
+
+        },
+    }))`
+    display:inline-block;
+    white-space: nowrap;
+`
```

Comment:
Not sure why you are using this `attrs` syntax. This can be simplified to something like

```suggestion
const Container = styled.div`
  display:inline-block;
  white-space: nowrap;
 margin:         `${({fontSize) => fontSize * 0.5 || 0.35 }rem`,
 padding:        `-${({fontSize) => fontSize * 0.5 || 0.25 }rem`,
`
```

Reply:
I did that before but I had too many warnings from styled-components saying I have too many rules being generated and to use this approach. It was very specific, so i changed it everywhere.